athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/serde/ObjectMapperUtil.java

```diff
@@ -59,4 +61,48 @@ else if (object instanceof FederationResponse) {
             throw new RuntimeException(ex);
         }
     }
+
+    public static <T> void assertSerializationBackwardsCompatible(Object object)
+    {
+        Class<?> clazz = object.getClass();
+        if (object instanceof FederationRequest)
+            clazz = FederationRequest.class;
+        else if (object instanceof FederationResponse) {
+            clazz = FederationResponse.class;
+        }
+        try (BlockAllocator allocator = new BlockAllocatorImpl()){
+            // check SerDe write, SerDe read
+            ByteArrayOutputStream serDeOut = new ByteArrayOutputStream();
+            ObjectMapper serDeV3 = VersionedObjectMapperFactory.create(allocator, SERDE_VERSION_THREE);
+            ObjectMapper serDeV2 = VersionedObjectMapperFactory.create(allocator, SERDE_VERSION_TWO);
+            serDeV3.writeValue(serDeOut, object);
+            byte[] serDeOutput = serDeOut.toByteArray();
+            assertEquals(object, serDeV2.readValue(new ByteArrayInputStream(serDeOutput), clazz));
+        }
+        catch (IOException | AssertionError ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public static <T> void assertSerializationForwardsCompatible(Object object)
+    {
+        Class<?> clazz = object.getClass();
+        if (object instanceof FederationRequest)
+            clazz = FederationRequest.class;
+        else if (object instanceof FederationResponse) {
+            clazz = FederationResponse.class;
+        }
+        try (BlockAllocator allocator = new BlockAllocatorImpl()){
+            // check SerDe write, SerDe read
+            ByteArrayOutputStream serDeOut = new ByteArrayOutputStream();
+            ObjectMapper serDeV3 = VersionedObjectMapperFactory.create(allocator, SERDE_VERSION_THREE);
+            ObjectMapper serDeV2 = VersionedObjectMapperFactory.create(allocator, SERDE_VERSION_TWO);
+            serDeV2.writeValue(serDeOut, object);
```

Comment:
Same comment about testing versions 1 and 3

Reply:
actually nvm, so I took a look at the code and the old PR for V2 and it looks like theres never a case where V1 can write/read and V2/V3 can write/read. 

I added the V2/V3 test because I wanted to confirm that if we don't add server side changes, that the connectors would still work as expected. 

When V2 was added, they did not push the connector changes until V2 was deployed serverside because the changes were not compatible.